i6_models/parts/conformer/frontend.py

```diff
@@ -0,0 +1,282 @@
+        self.check_valid()
+
+
+class ConformerVGGFrontendV1(nn.Module):
+    """
+    Convolutional Front-End
+
+    Uses explicit padding for ONNX exportability, see:
+    https://github.com/pytorch/pytorch/issues/68880
+    """
+
+    def __init__(self, model_cfg: ConformerVGGFrontendV1Config):
+        """
+        :param model_cfg: model configuration for this module
+        """
+        super().__init__()
+
+        model_cfg.check_valid()
+
+        conv_padding = self._get_padding(model_cfg.conv_kernel_size)
+        pool_1_padding = self._get_padding(model_cfg.pool_1_kernel_size)
+        pool_2_padding = self._get_padding(model_cfg.pool_2_kernel_size)
+
+        self.conv1 = nn.Conv2d(
+            in_channels=model_cfg.features,
+            out_channels=model_cfg.channels_conv_1_2,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv2 = nn.Conv2d(
+            in_channels=model_cfg.channels_conv_1_2,
+            out_channels=model_cfg.channels_conv_1_2,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.pooling1 = nn.MaxPool2d(
+            kernel_size=model_cfg.pool_1_kernel_size,
+            stride=model_cfg.pool_1_strides,
+            padding=pool_1_padding,
+        )
+        self.conv3 = nn.Conv2d(
+            in_channels=model_cfg.channels_conv_1_2,
+            out_channels=model_cfg.channels_conv_3_4,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv4 = nn.Conv2d(
+            in_channels=model_cfg.channels_conv_3_4,
+            out_channels=model_cfg.channels_conv_3_4,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.pooling2 = nn.MaxPool2d(
+            kernel_size=model_cfg.pool_2_kernel_size,
+            stride=model_cfg.pool_2_strides,
+            padding=pool_2_padding,
+        )
+        self.activation = model_cfg.activation
+
+    def forward(self, tensor: torch.Tensor) -> torch.Tensor:
+        """
+        T might be reduced to T' depending on ...
+
+        :param tensor: input tensor of shape [B,T,F]
+        :return: torch.Tensor of shape [B,T',F]
+        """
+        # conv 2d layers expect shape [B,F,T,C] so we have to transpose here
+        tensor = torch.transpose(tensor, 1, 2)  # [B,F,T]
+        # and add a dim
+        tensor = tensor[:, :, :, None]  # [B,F,T,C]
+        tensor = self.conv1(tensor)
+        tensor = self.conv2(tensor)
+        tensor = torch.transpose(tensor, 1, 2)  # transpose back to [B,T,F,C]
+        tensor = torch.squeeze(tensor)  # [B,T,F]
+
+        tensor = self.activation(tensor)
+        tensor = self.pooling1(tensor)
+
+        # conv 2d layers expect shape [B,F,T,C] so we have to transpose here
+        tensor = torch.transpose(tensor, 1, 2)  # [B,F,T]
+        tensor = tensor[:, :, :, None]  # [B,F,T,C]
```

Comment:
Again see my other comments. I think the comment is wrong, but also the `transpose`, and adding the dim here. All this should not be needed when you don't use the `squeeze` above, which is also wrong, I think.
```suggestion
```